compiler-rt/lib/rtsan/tests/rtsan_test_interceptors.cpp

```diff
@@ -0,0 +1,516 @@
+//===--- rtsan_test_interceptors.cpp - Realtime Sanitizer -------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
```

Comment:
In my experience LIT tests in are easier to maintain in LLVM environment

`TEST(TestRtsan,` are internals test, so probably fine to keep as GTEST

However `TEST(TestRtsanInterceptors` are likely better as regular LIT tests

Reply:
Actually even TestRtsan mostly can be done as LIT

BTW. Why I don't like GTEST in LLVM?

LIT is easier to reproduce and experiment and debug than with a particular test case of.

Also I don't trust incremental builds with GTEST, some times I am not sure that my changes are used in tests.

A couple of time CMakes bugs caused compiler-rt GTEST not executed on bots and it was unnoticed for a long time. LIT is majority, so unlikely it could happen with them.

Reply:
Seems reasonable to convert these to lit! I propose leaving this for this commit, and we convert them in the near future. Specifically this next commit will introduce the -fsanitize=realtime flag, and come with a bunch of the lit tests that we have written, so it will be easier to convert them once that infrastructure has landed.

One clarifying question, for when that work is done:

Should we have a lit test for every intereceptor? The reason we did it in gtest unit tests is that adding a new interceptor's test is a few lines, as opposed to a  brand new file with the lit boilerplate. This makes it easy to make sure that every interceptor does what we think it does. We worried about the "cost" of the new file for every interceptor.  In our demo branch, as we have it laid out, the lit tests test on the full end-to-end system level, while these tests actually hit every interceptor.

Reply:
> Should we have a lit test for every intereceptor? 

I don't have strong preference. For other sanitizer default interceptor behavior is to crash, so we usually create a test per interceptor.

> The reason we did it in gtest unit tests is that adding a new interceptor's test is a few lines, as opposed to a brand new file with the lit boilerplate. This makes it easy to make sure that every interceptor does what we think it does. We worried about the "cost" of the new file for every interceptor. In our demo branch, as we have it laid out, the lit tests test on the full end-to-end system level, while these tests actually hit every interceptor.

Yes, boilerplate of LIT is not nice. So there is a trade off. I guess you can wait and see.